compiler/src/iree/compiler/Dialect/Util/Analysis/Constant/ConstExpr.cpp

```diff
@@ -309,6 +309,32 @@ void ConstExprHoistingPolicy::initialize() {
   }
 }
 
+static bool doesHoistingIncreaseSizeSignificantly(
+    const ConstExprAnalysis::ConstValueInfo *info) {
+
+  int64_t inSizeBits = 0;
+  for (Value root : info->roots) {
+    // TODO: Are there any other types we care about here?
+    if (auto type = dyn_cast<ShapedType>(root.getType())) {
+      inSizeBits += type.getNumElements() * type.getElementTypeBitWidth();
```

Comment:
use `IREE::Util::getRoundedPhysicalStorageSize` instead of computing this yourself - it takes a shaped type and gets you the byte count and properly handles sub-byte types (here and below)